lib/SILOptimizer/Mandatory/TFUtilities.cpp

```diff
@@ -1621,10 +1621,18 @@ canonicalizeOperands(GraphGlobalConfiguration &configuration) {
 /// exist.
 StringRef GraphOperationInfo::getDeviceString() const {
   auto attr = inst->getAttribute(DEVICE_ATTR);
-  assert(attr.hasValue() && "Tensor op instruction has no device string");
+  assertWithDump(attr.hasValue(), "Tensor op instruction has no device string");
   return attr.getValue().getStringValue();
 }
 
+void GraphOperationInfo::assertWithDump(bool cond,
```

Comment:
The body of this function is not gated by a macro check, which would make it run even in no-assert mode. Is there a reason why this is not gated?

Reply:
Added the gate. Thanks.